src/main.rs

```diff
@@ -1,4 +1,143 @@
-fn main() {
-    println!("Hello, world!");
+use axum::{
+    async_trait,
+    extract::{FromRequest, Path, RequestParts},
+    handler::Handler,
+    http::StatusCode,
+    response::{IntoResponse, Json, Response},
+    routing::get,
+    Router, Server,
+};
+use serde_json::Value;
+use std::collections::HashMap;
+
+#[derive(Debug)]
+enum Version {
+    V1,
+}
+
+#[derive(Debug)]
+struct JsonProxyError(reqwest::Error);
+
+#[tokio::main]
+async fn main() {
+    let meta_routes = Router::new()
+        .route("/", get(root_handler))
+        .route("/ping", get(heartbeat_handler));
+
+    let transit_routes = Router::new()
+        .route("/bus", get(transit_bus_handler))
+        .route("/modes", get(transit_modes_handler));
+
+    let api_routes = Router::new()
+        .route("/contacts", get(contacts_handler))
+        .route("/dictionary", get(dictionary_handler))
+        .route("/faqs", get(faqs_handler))
+        .route("/food/named/menu/pause", get(pause_menu_handler))
+        .route("/printing/color-printers", get(color_printers_handler))
+        .route("/spaces/hours", get(hours_handler))
+        .route("/tools/help", get(help_handler))
+        .nest("/transit", transit_routes)
+        .route("/webcams", get(webcams_handler));
+
+    let app = Router::new()
+        .nest("/", meta_routes)
+        .nest("/api/:version", api_routes)
+        .fallback(fallback.into_service());
+
+    Server::bind(&"0.0.0.0:3000".parse().unwrap())
+        .serve(app.into_make_service())
+        .await
+        .expect("server failed to exit successfully");
+}
+
+#[allow(clippy::unused_async)]
+async fn root_handler() -> Result<&'static str, StatusCode> {
+    Ok("Hello world!")
+}
+
+#[allow(clippy::unused_async)]
+async fn heartbeat_handler() -> Result<&'static str, StatusCode> {
+    Ok("pong")
+}
+
+#[allow(clippy::unused_async)]
+async fn fallback() -> impl IntoResponse {
+    (StatusCode::NOT_FOUND, "Not Found".to_string())
+}
+
+async fn gh_pages_handler(filename: &str) -> Result<Json<Value>, JsonProxyError> {
+    let url = format!("https://stodevx.github.io/AAO-React-Native/{}", filename).to_string();
+    let resp = request_handler(&url).await?;
+    Ok(resp)
 }
```

Comment:
In coordination with my suggested change to `request_handler`, now you can make this a generic as well:

```rust
async fn gh_pages_handler<T>(filename: &str) -> Result<Json<T>, JsonProxyError>
where
    T: DeserializeOwned,
{
    let url = format!("https://stodevx.github.io/AAO-React-Native/{}", filename).to_string();
    let resp = request_handler(&url).await?;
    Ok(resp)
}
```